### PR TITLE
Reinstate original version of projectOnPlaneToRef with small amendment

### PR DESCRIPTION
**Pull request by BabylonJSGuide**
Ref https://forum.babylonjs.com/t/project-3d-point-on-plane/9862/18?u=johnk

### CODE REVIEW

**Comment by BabylonJSGuide** [comment]
Snapshot PG test https://playground.babylonjs.com/?snapshot=refs/pull/12827/merge#QD54M6#3

**Comment by RaananW** [comment]
t ixing n isu ith https://github.com/BabylonJS/Babylon.js/pull/12663R https://github.com/BabylonJS/Babylon.js/pull/12663?

**Comment by RaananW** [comment]
Is that a fix for #12663 ?

**Review by sebavan — APPROVED** [review]
@BabylonJSGuide the unit tests shoudl probably be changed as well to adapt to the new "old" code :-)

**Comment by BabylonJSGuide** [comment]
> Is that a fix for #12663 ?

It is.

File: packages/dev/core/src/Maths/math.vector.ts [hunk 1]
```diff
@@ -1144,48 +1144,34 @@ export class Vector3 {
...
 
-        // Calculate the length along the direction vector to the hit point
-        const hitDistance = Vector3.Dot(relativeOrigin, inverseNormal) / dotProduct;
+        const denom = Vector3.Dot(V, n);
 
-        // Apply the hit point by adding the direction scaled by the distance to the origin
-        result.set(origin.x + x * hitDistance, origin.y + y * hitDistance, origin.z + z * hitDistance);
+        //When the ray is close to parallel to the plane return infinity vector
+        if (Math.abs(denom) < Math.pow(10, -10)) {
+            origin.addToRef(new Vector3(Infinity, Infinity, Infinity), result);
```
[the remark this answers is not in the document]
Comment (BabylonJSGuide) [reply packages/dev/core/src/Maths/math.vector.ts:1167]: Possibly, I will check but will not be until tomorrow.

**Comment by BabylonJSGuide** [comment]
> @BabylonJSGuide the unit tests shoudl probably be changed as well to adapt to the new "old" code :-)

I am a bit confused about new system. Need some instructions on how to deal with the two failed checks.

**Comment by sebavan** [comment]
You can go in this file https://github.com/BabylonJS/Babylon.js/pull/12663/files#diff-7ad717d6f28a942a7f6dbc8c90e8071a0994061fd11be3a1b4bb4aeeff2740d0 and change the values to be covered by a unit test ?

**Comment by sebavan** [comment]
@BabylonJSGuide I ll do it for you :-)

**Comment by BabylonJSGuide** [comment]
> @BabylonJSGuide I ll do it for you :-)

Thank you

**Comment by sebavan** [comment]
Closing in favor of https://github.com/BabylonJS/Babylon.js/pull/12831 to have it in before the npm build, Thanks a lot @BabylonJSGuide

**Pull request closed without merging**